### PR TITLE
CMake: Update precompiled header and fix support

### PR DESCRIPTION
**Pull request by peterbell10**
This fixes the `USE_PRECOMPILED_HEADERS` cmake version check which was accidentally inverted, so it was always disabled.

I've also made the precompiled header so it only includes headers used in 95% or more of code, weighted by compile time. This limits it to the standard library, `c10` and a limited subset of `ATen/core`. Crucially, the new pch doesn't depend on `native_functions.yaml` so won't cause as much unnecessary rebuilding.

### CODE REVIEW

**Comment by dagitses** [comment]
> This fixes the `USE_PRECOMPILED_HEADERS` cmake version check which was accidentally inverted, so it was always disabled.
> 
> I've also made the precompiled header so it only includes headers used in 95% or more of code, weighted by compile time. This limits it to the standard library, `c10` and a limited subset of `ATen/core`. Crucially, the new pch doesn't depend on `native_functions.yaml` so won't cause as much unnecessary rebuilding.

great catch! That explains why I found zero impact on compile time when using this flag.

File: aten/src/ATen/core/ATen_pch.h [hunk 1]
```diff
@@ -0,0 +1,148 @@
+// This global header must not depend on native_functions.yaml or
```
Comment (dagitses) [inline aten/src/ATen/core/ATen_pch.h:1]: is there a particular sort order to this file? was this file generated?

can you add some comments about how this was created?

**Review by dagitses — APPROVED** [review]

File: aten/src/ATen/core/ATen_pch.h (hunk 1 above)
Comment (peterbell10) [reply aten/src/ATen/core/ATen_pch.h:1]: I've published my scripts in https://github.com/peterbell10/build_analysis/ and added a comment for how to recreate it.

**Comment by dagitses** [comment]
can you rebase this please?

**Comment by peterbell10** [comment]
@dagitses rebased on `viable/strict`.

**Pull request closed without merging**